servicebox-app/src/test/java/io/halkyon/ServiceDiscoveryJobTest.java

```diff
@@ -0,0 +1,82 @@
+package io.halkyon;
+
+import static io.halkyon.utils.TestUtils.createCluster;
+import static io.halkyon.utils.TestUtils.createService;
+import static io.restassured.RestAssured.given;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.awaitility.Awaitility.await;
+import static org.hamcrest.CoreMatchers.is;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.argThat;
+import static org.mockito.ArgumentMatchers.eq;
+
+import javax.inject.Inject;
+import javax.ws.rs.core.MediaType;
+
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+import io.halkyon.model.Cluster;
+import io.halkyon.model.Service;
+import io.halkyon.services.KubernetesClientService;
+import io.halkyon.services.ServiceDiscoveryJob;
+import io.halkyon.utils.ClusterNameMatcher;
+import io.quarkus.scheduler.Scheduler;
+import io.quarkus.test.junit.QuarkusTest;
+import io.quarkus.test.junit.mockito.InjectMock;
+
+@QuarkusTest
+public class ServiceDiscoveryJobTest {
+
+    @InjectMock
+    KubernetesClientService mockKubernetesClientService;
+
+    @Inject
+    ServiceDiscoveryJob job;
+
+    @Inject
+    Scheduler scheduler;
+
+    @Test
+    public void testJobShouldMarkClaimAsErrorAfterMaxAttemptsExceeded(){
+        pauseScheduler();
+        Service service = createService("ServiceDiscoveryJobTest", "any", "host:1111");
+
+        // When we run the job once:
+        // Then:
+        // - the service "ServiceDiscoveryJobTest" should keep the deployed flag to false because it's not deployed in any cluster yet
+        // When we install the service in a cluster
```

Comment:
Can this test really work as currently we only run/test primaza on a local machine or using github but without a kind cluster ?

Reply:
Yes, we're mocking the KubernetesClient.